homeassistant/components/nasweb/coordinator.py

```diff
@@ -0,0 +1,132 @@
+"""Message routing coordinators for handling NASweb push notifications."""
+import asyncio
+from datetime import timedelta
+import logging
+import time
+from typing import Any, Optional
+
+from aiohttp import web
+from webio_api import Output as NASwebOutput, WebioAPI
+from webio_api.const import KEY_DEVICE_SERIAL, KEY_OUTPUTS, KEY_TYPE, TYPE_STATUS_UPDATE
+
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+from homeassistant.helpers.update_coordinator import DataUpdateCoordinator, UpdateFailed
+
+from .const import STATUS_UPDATE_MAX_TIME_INTERVAL
+from .relay_switch import RelaySwitch
+
+_LOGGER = logging.getLogger(__name__)
+
+
+class NotificationCoordinator:
+    """Coordinator redirecting push notifications for this integration to appropriate NASwebCoordinator."""
+
+    def __init__(self) -> None:
+        """Initialize coordinator."""
+        self._coordinators: dict[str, NASwebCoordinator] = {}
+
+    def add_coordinator(self, serial: str, cord: "NASwebCoordinator") -> None:
+        """Remove NASwebCoordinator to possible notification targets."""
+        self._coordinators[serial] = cord
+        _LOGGER.debug("Added NASwebCoordinator for NASweb[%s]", serial)
+
+    def remove_coordinator(self, serial: str) -> None:
+        """Remove NASwebCoordinator from possible notification targets."""
+        if self._coordinators.pop(serial, None) is None:
+            _LOGGER.warning("Trying to remove nonexistent NASwebCoordinator")
+        _LOGGER.debug("Removed NASwebCoordinator for NASweb[%s]", serial)
+
+    def has_coordinators(self) -> bool:
+        """Check if there is any registered coordinator for push notifications."""
+        return len(self._coordinators) > 0
+
+    async def check_connection(self, serial: str) -> bool:
+        """Wait for first status update to confirm connection with NASweb."""
+        nasweb_coordinator = self._coordinators.get(serial)
+        if nasweb_coordinator is None:
+            _LOGGER.error("Cannot check connection. No device match serial number")
+            return False
+        counter = 0
+        _LOGGER.debug("Checking connection with: %s", serial)
+        while not nasweb_coordinator.is_connection_confirmed() and counter < 10:
+            await asyncio.sleep(1)
+            counter += 1
+            _LOGGER.debug("Checking connection with: %s (%s)", serial, counter)
+        return nasweb_coordinator.is_connection_confirmed()
+
+    async def handle_notification(self, request: web.BaseRequest) -> web.Response:
+        """NASweb push notification handler."""
+        if not self.has_coordinators():
+            return web.Response()
```

Comment:
Why does this always return an empty web.Response, why not just return None?

Reply:
I was following some example where that was the case, assumed that it's default expected return, will change to return None